include/query.php

```diff
@@ -174,6 +185,20 @@ public function filter_query( $lang ) {
 				}
 			}
 		} else {
+			// Set the language correctly in the query, since WordPress merges the language with the other query vars when the operator is OR.
+			if ( isset( $this->query->tax_query->queried_terms['language'] ) && 'all' !== $this->query->tax_query->queried_terms['language']['terms'] ) {
+				$langs = $this->query->tax_query->queried_terms['language']['terms'];
+				if ( is_string( $langs ) ) {
+					$langs = explode( ',', $langs );
+				}
+				$langs = array_map( array( $this->model, 'get_language' ), $langs );
+				$langs = array_filter( $langs );
+				unset( $qvars['lang'] );
```

Comment:
Do we ne need to unset?

Reply:
Thought it would be necessary, but you're right it's not.